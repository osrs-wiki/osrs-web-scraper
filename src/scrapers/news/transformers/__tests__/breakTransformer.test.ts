import {
  MediaWikiBreak,
  MediaWikiBuilder,
  MediaWikiContent,
  MediaWikiHeader,
  MediaWikiText,
  MediaWikiTOC,
} from "../../../../utils/mediawiki";
import NewsBreakTransformer from "../breakTransformer";

describe("NewsBreakTransformer", () => {
  it("should combine three lines breaks to one", () => {
    const originalContent: MediaWikiContent[] = [
      new MediaWikiTOC(),
      new MediaWikiBreak(),
      new MediaWikiBreak(),
      new MediaWikiBreak(),
      new MediaWikiText("The Old School Team.", { italics: true }),
    ];
    const transformed = new NewsBreakTransformer().transform(originalContent);
    expect(
      new MediaWikiBuilder().addContents(transformed).build()
    ).toMatchSnapshot();
  });

  it("should remove the line breaks after a header", () => {
    const originalContent: MediaWikiContent[] = [
      new MediaWikiTOC(),
      new MediaWikiBreak(),
      new MediaWikiHeader("Test", 2),
      new MediaWikiBreak(),
      new MediaWikiBreak(),
      new MediaWikiText("The Old School Team.", { italics: true }),
    ];
    const transformed = new NewsBreakTransformer().transform(originalContent);
    expect(
      new MediaWikiBuilder().addContents(transformed).build()
    ).toMatchSnapshot();
  });
});
