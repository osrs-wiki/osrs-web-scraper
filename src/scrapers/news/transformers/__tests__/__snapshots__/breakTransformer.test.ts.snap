// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsBreakTransformer should combine three lines breaks to one 1`] = `
"__TOC__

''The Old School Team.''"
`;

exports[`NewsBreakTransformer should remove the line breaks after a header 1`] = `
"__TOC__

==Test==
''The Old School Team.''"
`;
